packages/react-sdk/src/client.ts

```diff
@@ -0,0 +1,595 @@
+    )
+  }
+
+  onReady(config: { timeout?: number } = {}): Promise<OnReadyResult> {
+    let timeoutId: number | undefined
+    let timeout: number = DEFAULT_ON_READY_TIMEOUT
+    if (config && config.timeout !== undefined) {
+      timeout = config.timeout
+    }
+
+    const timeoutPromise = new Promise<OnReadyResult>(resolve => {
+      timeoutId = setTimeout(() => {
+        resolve({
+          success: false,
+          reason:
+            'failed to initialize onReady before timeout, either the datafile or user info was not set before the timeout',
+        })
+      }, timeout) as any
+    })
+
+    return Promise.race([this.dataReadyPromise, timeoutPromise]).then(res => {
+      clearTimeout(timeoutId)
+      return res
+    })
+  }
+
+  setUser(userInfo: { id?: string; attributes?: { [key: string]: any } }): void {
+    // TODO add check for valid user
+    if (userInfo.id) {
+      this.user.id = userInfo.id
+    }
+    if (userInfo.attributes) {
+      this.user.attributes = userInfo.attributes
+    }
+    if (!this.isUserPromiseResolved) {
+      this.userPromiseResovler(this.user)
+      this.isUserPromiseResolved = true
+    }
+    this.onUserUpdateHandlers.forEach(handler => handler(this.user))
+  }
+
+  onUserUpdate(handler: OnUserUpdateHandler): DisposeFn {
+    this.onUserUpdateHandlers.push(handler)
+
+    return () => {
+      const ind = this.onUserUpdateHandlers.indexOf(handler)
+      if (ind > -1) {
+        this.onUserUpdateHandlers.splice(ind, 1)
+      }
+    }
+  }
+
+  /**
+   * @param {string} experimentKey
+   * @param {string} userId
+   * @param {UserAttributes} [attributes]
+   * @returns {(string | null)}
+   * @memberof OptimizelySDKWrapper
+   */
+  public activate(
+    experimentKey: string,
+    overrideUserId?: string,
+    overrideAttributes?: UserAttributes,
+  ): string | null {
+    const [userId, userAttributes] = this.getUserIdAndAttributes(
+      overrideUserId,
+      overrideAttributes,
+    )
+    if (userId === null) {
+      logger.info(
+        'Not activating experiment "%s" because userId is not set',
+        experimentKey,
+      )
+      return null
+    }
+    return this.client.activate(experimentKey, userId, userAttributes)
+  }
+
+  /**
+   *
+   *
+   * @param {string} experimentKey
+   * @param {string} userId
+   * @param {UserAttributes} [attributes]
+   * @returns {(string | null)}
+   * @memberof OptimizelySDKWrapper
+   */
+  public getVariation(
+    experimentKey: string,
+    overrideUserId?: string,
+    overrideAttributes?: UserAttributes,
+  ): string | null {
+    const [userId, userAttributes] = this.getUserIdAndAttributes(
+      overrideUserId,
+      overrideAttributes,
+    )
+    if (userId === null) {
+      logger.info(
+        'getVariation returned null for experiment "%s" because userId is not set',
+        experimentKey,
+      )
+      return null
+    }
+    return this.client.getVariation(experimentKey, userId, userAttributes)
+  }
+
+  track(
```

Comment:
This should be public with doc comments

Reply:
Should we delete all `public`, since `public` is the default?